@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,182 @@ export namespace MapViewUtils {
         if (mapTargetWorld === null) {
             return;
         }
-        const mapTargetNormal = projection.surfaceNormal(mapTargetWorld, new THREE.Vector3());
 
-        camera.updateMatrixWorld(true);
-        const dir = cache.vector3[1].setFromMatrixColumn(camera.matrixWorld, 2);
-        const up = cache.vector3[2].setFromMatrixColumn(camera.matrixWorld, 1);
+        applyAzimuthAroundTarget(mapView, rotationTargetWorld, -deltaAzimuth);
 
-        // Due to inaccuracies it can happen that the tilt angle gets less than 0.
-        // Using the dot product of the direction vector and the surface normal alone does not give us the sign.
-        const tiltSign = Math.sign(up.dot(mapTargetNormal));
+        const clampedDeltaTilt = computeClampedDeltaTilt(
+            mapView,
+            offsetX,
+            offsetY,
+            deltaTilt,
+            maxTiltAngle,
+            mapTargetWorld,
+            rotationTargetWorld
+        );
 
-        const currentTilt = (tiltSign === 0 ? 1 : tiltSign) * dir.angleTo(mapTargetNormal);
+        applyTiltAroundTarget(mapView, rotationTargetWorld, clampedDeltaTilt);
+    }
 
-        //FIXME(HARP-11926): For globe tilt in the map center is different from the tilt in the rotation center,
-        //hence the clamped tilt is too conservative.
-        const clampedDeltaTilt =
-            MathUtils.clamp(deltaTilt + currentTilt, 0, maxTiltAngle) - currentTilt;
-        if (Math.abs(clampedDeltaTilt) <= Number.EPSILON) {
-            return;
+    /**
+     * @hidden
+     * @internal
+     *
+     * Applies the given Azimith to the camera around the supplied target.
+     */
+    function applyAzimuthAroundTarget(
+        mapView: MapView,
+        rotationTargetWorld: THREE.Vector3,
+        deltaAzimuth: number
+    ) {
+        const camera = mapView.camera;
+        const projection = mapView.projection;
+
+        const headingAxis = projection.surfaceNormal(rotationTargetWorld, cache.vector3[0]);
+        const headingQuat = cache.quaternions[0].setFromAxisAngle(headingAxis, deltaAzimuth);
+        camera.quaternion.premultiply(headingQuat);
+        camera.position.sub(rotationTargetWorld);
+        camera.position.applyQuaternion(headingQuat);
+        camera.position.add(rotationTargetWorld);
+    }
+
+    /**
+     * @hidden
+     * @internal
+     *
+     * Clamps the supplied `deltaTilt` to the `maxTiltAngle` supplied. Note, when a non-zero offset
+     * is applied, we apply another max angle of 89 degrees to the rotation center to prevent some
+     * corner cases where the angle at the rotation center is 90 degrees and therefore intersects
+     * the geometry with the near plane.
+     */
+    function computeClampedDeltaTilt(
+        mapView: MapView,
+        offsetX: number,
+        offsetY: number,
+        deltaTilt: number,
+        maxTiltAngle: number,
+        mapTargetWorld: THREE.Vector3,
+        rotationTargetWorld: THREE.Vector3
```

Comment:
This function doesn't need the offsets anymore. It only needs to know whether the rotation target is the same as the map target. You can make rotationTargetWorld optional. If undefined, the rotation has mapTargetWorld as center.

Reply:
I need them to check whether the points are the same or not, I thought it would be better to check the offset rather than do an exact check, i.e. `mapTargetWorld.equals(rotationTargetWorld)`, because of potential rounding issues? Though I guess they would be the same if the offsets are the same, so maybe I should just go with the `equals`